### PR TITLE
[server 0.1036.5000] Bump server version and update deps to address CVEs

### PR DESCRIPTION
**Pull request by alexvy86**
## Description

- Update dependencies in server npm packages to address [CVE-2023-45857](https://github.com/advisories/GHSA-wf5p-g6vw-rhxx) and [CVE-2024-21484](https://github.com/advisories/GHSA-rh63-9qcf-83gf).
  - In main we were already using axios 1.x so we addressed the CVE by updating to 1.6, but since they also release a 0.28 version that addresses the CVE, it seemed safer to only bump from 0.26 to 0.28 here.
- Bump of the server version to 0.1036.5002 which didn't happen after publishing 0.1036.5001.
- Cherry pick of https://github.com/alexvy86/FluidFramework/commit/a0368d65fcbc83f20ac6646a393bea9f1f602ccc so this branch can work on our CI infrastructure which has gone through updates.
- Cherry pick of several commits that add permissions to github workflows which are now required.

Follow up work is to publish the npm packages with these fixes, then do a patch release of azure-client 1.1.x that updates its dependencies on them to this latest version.

## Reviewer Guidance

The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

### CODE REVIEW

**Comment by CraigMacomber** [comment]
I'm unclear on why this needs publishing. As far as I can tell the change is only in the lock file, which is not involved at all in the produced package. If all you do is update the lock file the new package will be the same as the old one, and any updates you did will still have to be done by the users of that package, and thet could have done them themselves without us publishing new packages.

If we also publish some packed version of the packages including the deps, this seems like a no-op, other than updating the versions that will get used on CI for future PRs (which is a good thing, but does not need releasing).

File: server/routerlicious/lerna-package-lock.json [hunk 1]
```diff
@@ -267,7 +267,7 @@
 		"@dsherret/to-absolute-glob": {
 			"version": "2.0.2",
 			"resolved": "https://registry.npmjs.org/@dsherret/to-absolute-glob/-/to-absolute-glob-2.0.2.tgz",
-			"integrity": "sha512-InCaQ/KEOcFtAFztn47wadritBLP2nT6m/ucbBnIgI5YwxuMzKKCHtqazR2+D1yR6y1ZTnPea9aLFEUrTttUSQ==",
+			"integrity": "sha1-H2R13IvZdM6gei2vOGSzF7HdMyw=",
```
[the remark this answers is not in the document]
Comment (CraigMacomber) [reply server/routerlicious/lerna-package-lock.json:270]: If I remember my annoying lerna/npm bugs from long ago correctly, I think the fix is to clear some caches, revert this file, and install again. I'm not sure though.

[the remark this answers is not in the document]
Comment (alexvy86) [reply server/routerlicious/lerna-package-lock.json:270]: Yeah, I called it out in the description and put the hash changes in their own commit to try to simplify that. I guess I can revert that commit after bumping the deps that the PR really cares about, trying that.

File: server/routerlicious/packages/services-client/src/restWrapper.ts [hunk 2]
```diff
@@ -129,7 +129,7 @@ export class BasicRestWrapper extends RestWrapper {
         return new Promise<T>((resolve, reject) => {
             this.axios.request<T>(options)
                 .then((response) => { resolve(response.data); })
-                .catch((error: AxiosError) => {
+                .catch((error: AxiosError<any>) => {
```
Comment (alexvy86) [inline server/routerlicious/packages/services-client/src/restWrapper.ts:132]: Same change we applied [when upgrading to Axios 1.x in main](https://github.com/microsoft/FluidFramework/pull/17419/files#diff-6ad2b284866501b71c805cb9805e1077821ab8172b0c513abe1fe8d90fdf0b30R172).

File: experimental/PropertyDDS/packages/property-common/package.json [hunk 3]
```diff
@@ -82,7 +82,7 @@
     "cross-env": "^7.0.2",
     "eslint": "~8.6.0",
     "mocha": "^10.0.0",
-    "nock": "^10.0.1",
+    "nock": "^13.3.3",
```
Comment (CraigMacomber) [inline experimental/PropertyDDS/packages/property-common/package.json:85]: it seems odd to update a bunch of non server things on a server release branch.

Comment (alexvy86) [reply experimental/PropertyDDS/packages/property-common/package.json:85]: I agree and was thinking how can we do this better. This is from cherry-picking a commit that fixed lts to be able to run most of our pipelines. While a good chunk of it was about client and azure, I was hoping not to have to disentangle that PR.

**Pull request closed without merging**